REL/wikipedia.py

```diff
@@ -1,138 +1,155 @@
-        """
-        Generates wiki id/name and name/id index.
-
-        :return: disambiguation index
-        """
-
-        wiki_id_name_map = {"ent_name_to_id": {}, "ent_id_to_name": {}}
-        path = os.path.join(self.base_url, "basic_data/wiki_name_id_map.txt")
-        with open(path, "r", encoding="utf-8") as f:
-            for line in f:
-                line = line.rstrip()
-                parts = line.split("\t")
-
-                ent_wiki_id = int(parts[1])
-                ent_name = unquote(parts[0])
-
-                if ent_wiki_id not in self.wiki_disambiguation_index:
-                    wiki_id_name_map["ent_name_to_id"][ent_name] = ent_wiki_id
-                    wiki_id_name_map["ent_id_to_name"][ent_wiki_id] = ent_name
-        return wiki_id_name_map
-
-    def generate_wiki_redirect_map(self):
-        """
-        Generates redirect index.
-
-        :return: redirect index
-        """
-        wiki_redirects_index = {}
-        wiki_redirects_id_index = {}
-        path = os.path.join(self.base_url, "basic_data/wiki_redirects.txt")
-
-        with open(path, "r", encoding="utf-8",) as f:
-            for line in f:
-                line = line.rstrip()
-                parts = line.split("\t")
-                if len(parts) < 2:
-                    continue
-                parts[1] = unquote(parts[1])
-                wiki_redirects_index[unquote(parts[0])] = parts[1]
-                if len(parts) == 3:
-                    wiki_redirects_id_index[int(parts[2])] = parts[1]
-        return wiki_redirects_index, wiki_redirects_id_index
+import os
+from urllib.parse import unquote
+
+from REL.utils import first_letter_to_uppercase, trim1
+
+"""
+Class responsible for loading Wikipedia files. Required when filling sqlite3 database with e.g. p(e|m) index.
+"""
+
+
+class Wikipedia:
+    def __init__(self, base_url, wiki_version):
+        self.base_url = base_url + wiki_version
+        # if include_wiki_id_name:
+        self.wiki_disambiguation_index = self.generate_wiki_disambiguation_map()
+        print("Loaded wiki disambiguation index")
+        (
+            self.wiki_redirects_index,
+            self.wiki_redirects_id_index,
+        ) = self.generate_wiki_redirect_map()
+        print("Loaded wiki redirects index")
+        self.wiki_id_name_map = self.gen_wiki_name_map()
+        print("Loaded entity index")
+
+    def preprocess_ent_name(self, ent_name):
+        """
+        Preprocesses entity name.
+
+        :return: Preprocessed entity name.
+        """
+        ent_name = ent_name.strip()
+        ent_name = trim1(ent_name)
+        ent_name = ent_name.replace("&amp;", "&")
+        ent_name = ent_name.replace("&quot;", '"')
+        ent_name = ent_name.replace("_", " ")
+        ent_name = first_letter_to_uppercase(ent_name)
+
+        ent_name = self.wiki_redirect_ent_title(ent_name)
+        return ent_name
+
+    def preprocess_ent_name_for_wiki_id_lookup(self, ent_name):
```

Comment:
This the new function I would which is relevant.

Reply:
So if I understand correctly, we were unknowingly doing "too much" preprocessing with our first iteration, which is why this new function is just a slimmed up version of `preprocess_ent_name`?

Is there a need to keep `preprocess_ent_name` after these changes are verified?